store/types/validity.go

```diff
@@ -1,15 +1,28 @@
 package types
 
-// AssertValidKey checks if the key is valid(key is not nil)
+var (
+	// 128K - 1
+	MaxKeyLength = (1 << 17) - 1
+	// 2G - 1
+	MaxValueLength = (1 << 31) - 1
```

Comment:
The default size of a LevelDB log file [is 2MB](https://github.com/google/leveldb/blob/main/doc/impl.md#log-files) and compaction occurs [at 4MB](https://github.com/google/leveldb/blob/main/doc/impl.md#level-0). The SDK [uses the default settings](https://github.com/cosmos/cosmos-db/blob/main/goleveldb.go#L28) when creating a GoLevelDB backend. It is a core assumption of LevelDB operations, like compaction, that these files will contain many key/val pairs. Many promises of the database, including but not limited to performance, rely on that assumption. This may not be a problem if there are only a handful of key/val pairs that are larger than 4MB, but if there are more than just a few, then it does become an issue.

Reply:
Large keys definitely hurt performance because they are usually part of the index structure to do binary search. I guess no one is actually using big keys in practice.
Maybe a better thing to do is set a smaller limit in sdk, but third-parties can enlarge it since it's a variable, it's safer than having a bigger limit in sdk while third-party setting a smaller one.
But we'll have more communication to do in this way, WDYT? @webmaster128 @testinginprod

Reply:
CosmWasm never allowed values greater than 128 KiB. I don't know how much wrapping overhead there is in wasmd (probably none) and iavl. But I guess any limit >= 256 KiB would not hurt us.

Reply:
> Large keys

We are talking about values, right?

Reply:
keys ;D  I think there's not less worries about large values

Reply:
The 2-4MB limits I discuss above are impacted by the total size of a key/val pair, not just the key size or val size in isolation. My main point is that if you want to increase either of those limits, or both of them, to something approaching those 2-4MB limits, then you would also need to increase the limits of LevelDB itself; increasing the former without changes to the latter would result in really bad outcomes.